_drafts/programatically-configured-bots.md

```diff
@@ -68,23 +68,28 @@ robot.adapter.once 'connected', ->
 robot.run()
 ```
 
-Open questions to answer:
-
-- should `loadScriptPackage` take a string to require, or expect a function? if it takes a function, that could be re-used more generally
-
 ## Detailed design
 
 > Describe the design of the solution in detail. The detail in this section should be sufficient for someone who is *not* one of the authors to be able to reasonably implement the feature.
 
-TODO
+Much of the logic for loading scripts is currently in `bin/hubot`. It should all be moved to `Robot`. You can see there are several `load` methods, `loadExternalScripts` which loads external scripts from packages, `loadScripts` which calls that plus loads scripts the `scripts` directory. 
```

Comment:
I agree. I was playing around with a branch last night, and it didn't seem as useful to have access to these function on the robot itself.